From 15ef6cc22a6fe5ff638f140fc0df37c9c168d516 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:10:25 +0200
Subject: [PATCH] tests: py: fix up udp csum fixup output

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 036a1eb3304995e4e000c552b9dfd33b1073ddb6

commit 036a1eb3304995e4e000c552b9dfd33b1073ddb6
Author: Florian Westphal <fw@strlen.de>
Date:   Wed Sep 11 14:23:01 2024 +0200

    tests: py: fix up udp csum fixup output

    Preceeding commit switched udp to use the inkernel csum parser, so tests
    warn:

    WARNING: line 7: 'add rule ip test-ip4 input iif "lo" udp checksum set 0':
    '[ payload write reg 1 => 2b @ transport header + 6 csum_type 1 csum_off 6 csum_flags 0x0 ]' mismatches
    '[ payload write reg 1 => 2b @ transport header + 6 csum_type 0 csum_off 0 csum_flags 0x1 ]'

    Fixes: f89abfb4068d ("proto: use NFT_PAYLOAD_L4CSUM_PSEUDOHDR flag to mangle UDP checksum")
    Signed-off-by: Florian Westphal <fw@strlen.de>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 tests/py/inet/udp.t.payload | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/py/inet/udp.t.payload b/tests/py/inet/udp.t.payload
index e6beda7..32f7f8c 100644
--- a/tests/py/inet/udp.t.payload
+++ b/tests/py/inet/udp.t.payload
@@ -236,7 +236,7 @@ inet test-inet input
   [ meta load l4proto => reg 1 ]
   [ cmp eq reg 1 0x00000011 ]
   [ immediate reg 1 0x00000000 ]
-  [ payload write reg 1 => 2b @ transport header + 6 csum_type 1 csum_off 6 csum_flags 0x0 ]
+  [ payload write reg 1 => 2b @ transport header + 6 csum_type 0 csum_off 0 csum_flags 0x1 ]
 
 # iif "lo" udp dport set 65535
 inet test-inet input
@@ -245,4 +245,4 @@ inet test-inet input
   [ meta load l4proto => reg 1 ]
   [ cmp eq reg 1 0x00000011 ]
   [ immediate reg 1 0x0000ffff ]
-  [ payload write reg 1 => 2b @ transport header + 2 csum_type 1 csum_off 6 csum_flags 0x0 ]
+  [ payload write reg 1 => 2b @ transport header + 2 csum_type 0 csum_off 0 csum_flags 0x1 ]
